module/VuFind/src/VuFind/Db/Table/OaiResumption.php

```diff
@@ -85,11 +86,51 @@ public function removeExpired()
      *
      * @param string $token The resumption token to retrieve.
      *
-     * @return ?\VuFind\Db\Row\OaiResumption
+     * @return     ?\VuFind\Db\Row\OaiResumption
+     * @deprecated Use OaiResumption::findWithId
      */
     public function findToken($token)
     {
-        return $this->select(['id' => $token])->current();
+        return $this->findWithId($token);
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key; return null if it
+     * is not found.
+     *
+     * @param string $id Id used for the search.
+     *
+     * @return ?\VuFind\Db\Row\OaiResumption
+     */
+    public function findWithId(string $id): ?OaiResumptionEntityInterface
+    {
+        return $this->select(['id' => $id])->current();
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key and where the token is null.
+     *
+     * @param string $id Id used for the search.
+     *
+     * @return ?\VuFind\Db\Row\OaiResumption
+     * @todo   In future, we should migrate data to prevent null token fields, which will make this method obsolete.
+     */
+    final public function findWithLegacyIdToken(string $id): ?OaiResumptionEntityInterface
+    {
+        return $this->select(['id' => $id, 'token' => null])->current();
```

Comment:
It might be worth validating the ID as an integer, since if you try to look up a non-integer value in an integer column, it causes a database error. And/or maybe we need to adjust the type of the `$id` argument here.

```suggestion
        return intval($id)> 0 ? $this->select(['id' => $id, 'token' => null])->current() : null;
```

What do you think?

Reply:
had to actually do a more specific check as intval('54adsdasd') return 54. Added also a small test, but creating tests for db entities is rather tricky.

Reply:
Also, changed the conversion to happen in service.